<a href="https://colab.research.google.com/github/kittimaxz/data_viz_2022/blob/main/Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# เตรียมข้อมูล

## นำเข้าข้อมูล

### ข้อมูลโควิด ที่มา : https://data.go.th/dataset/covid-19-daily

เป็นข้อมูลที่แสดงถึงข้อมูลของผู้ติดเชื้อตั้งแต่รายแรก จนถึงรายปัจจุบัน คือ วันที่ 12 มกราคม 2020 ถึง 12 มีนาคม 2022 โดยแสดงถึงวันที่ที่มีการติดเชื้อ เพศ ความเสี่ยง(ที่ติดเชื้อ) และจังหวัด 

***เพื่อต้องการรู้ว่าผู้ติดเชื้อนั้นเป็นเพศใดมากกว่ากัน และมีความเสี่ยงและจังหวัดที่ติดเชื้อมากที่สุดเป็นความเสี่ยงและจังหวัดใดบ้าง***

In [62]:
import pandas as pd # import ฟังก์ชัน pandasimport pandas as pd # import ฟังก์ชัน pandas

In [63]:
from google.colab import drive # เชื่อม drive ของเรา ถ้าเชื่อมสำเร็จจะขึ้นคำว่าMounted at /content/drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [64]:
import os # จัดการเกี่ยวกับไฟล์ต่างๆ ในโฟล์เดอร์

In [65]:
path = '/content/drive/My Drive/dataviz_2021_data' # ชี้ว่า data เราอยู่โฟล์เดอร์ไหน

### ข้อมูลโควิด ตั้งแต่วันที่ 12 มกราคม 2020 ถึง 11 สิงหาคม 2021

In [66]:
data_file_path1 = os.path.join(path,'confirmed-cases.csv') # เอาที่อยู่ของไฟล์เชื่อมกับที่อยู่ของไฟล์เรา
data_covid1 = pd.read_csv(data_file_path1)

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2882: DtypeWarning: Columns (4) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [67]:
data_covid1

No. announce_date Notified date   sex   age Unit nationality  \
0            1     12/1/2020           NaN  หญิง  61.0   ปี       China   
1            2     17/1/2020           NaN  หญิง  74.0   ปี       China   
2            3     22/1/2020           NaN  หญิง  73.0   ปี    Thailand   
3            4     22/1/2020           NaN   ชาย  68.0   ปี       China   
4            5     24/1/2020           NaN  หญิง  66.0   ปี       China   
...        ...           ...           ...   ...   ...  ...         ...   
839766  839767     12/8/2021     11/8/2021   ชาย  52.0   ปี         NaN   
839767  839768     12/8/2021     11/8/2021   ชาย  34.0   ปี    Thailand   
839768  839769     12/8/2021     11/8/2021   ชาย  31.0   ปี    Thailand   
839769  839770     12/8/2021     11/8/2021   ชาย  83.0   ปี    Thailand   
839770  839771     12/8/2021     11/8/2021   ชาย  47.0   ปี    Thailand   

       province_of_isolation                              risk  \
0              กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ   
1              กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ   
2                     นครปฐม  คนต่างชาติเดินทางมาจากต่างประเทศ   
3              กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ   
4                    นนทบุรี  คนต่างชาติเดินทางมาจากต่างประเทศ   
...                      ...                               ...   
839766             กาญจนบุรี                  ทัณฑสถาน/เรือนจำ   
839767             กาญจนบุรี                  ทัณฑสถาน/เรือนจำ   
839768             กาญจนบุรี                  ทัณฑสถาน/เรือนจำ   
839769             กาญจนบุรี                  ทัณฑสถาน/เรือนจำ   
839770             กาญจนบุรี                  ทัณฑสถาน/เรือนจำ   

       province_of_onset district_of_onset  
0          กรุงเทพมหานคร               NaN  
1          กรุงเทพมหานคร               NaN  
2                 นครปฐม             เมือง  
3          กรุงเทพมหานคร               NaN  
4          กรุงเทพมหานคร               NaN  
...                  ...               ...  
839766         กาญจนบุรี               NaN  
839767         กาญจนบุรี               NaN  
839768         กาญจนบุรี               NaN  
839769         กาญจนบุรี               NaN  
839770         กาญจนบุรี               NaN  

[839771 rows x 11 columns]

### ข้อมูลโควิด ตั้งแต่วันที่ 12 สิงหาคม 2021 ถึง 25 ตุลาคม 2021

In [68]:
data_file_path2 = os.path.join(path,'confirmed-cases-since-120864.csv')
data_covid2 = pd.read_csv(data_file_path2)

In [69]:
data_covid2

No. announce_date Notified date   sex   age Unit nationality  \
0         816990     12/8/2021     11/8/2021   ชาย   7.0   ปี    Thailand   
1         816991     12/8/2021     11/8/2021   ชาย   1.0   ปี    Thailand   
2         816992     12/8/2021     11/8/2021   ชาย  35.0   ปี    Thailand   
3         816993     12/8/2021     11/8/2021  หญิง  33.0   ปี    Thailand   
4         816994     12/8/2021     11/8/2021  หญิง  14.0   ปี    Thailand   
...          ...           ...           ...   ...   ...  ...         ...   
1042163  1859153    25/10/2021    24/10/2021   ชาย  45.0   ปี         NaN   
1042164  1859154    25/10/2021    24/10/2021   ชาย  31.0   ปี         NaN   
1042165  1859155    25/10/2021    24/10/2021   ชาย  31.0   ปี         NaN   
1042166  1859156    25/10/2021    24/10/2021   ชาย  24.0   ปี         NaN   
1042167  1859157    25/10/2021    24/10/2021   ชาย  25.0   ปี         NaN   

        province_of_isolation                                         risk  \
0                    เชียงราย                                        อื่นๆ   
1                    เชียงราย                                        อื่นๆ   
2                    เชียงราย                                        อื่นๆ   
3                    เชียงราย                                        อื่นๆ   
4                    เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
...                       ...                                          ...   
1042163         กรุงเทพมหานคร                             ทัณฑสถาน/เรือนจำ   
1042164         กรุงเทพมหานคร                             ทัณฑสถาน/เรือนจำ   
1042165         กรุงเทพมหานคร                             ทัณฑสถาน/เรือนจำ   
1042166         กรุงเทพมหานคร                             ทัณฑสถาน/เรือนจำ   
1042167             เชียงใหม่                             ทัณฑสถาน/เรือนจำ   

        province_of_onset district_of_onset  
0                     NaN               NaN  
1                     NaN               NaN  
2                     NaN               NaN  
3                     NaN               NaN  
4                     NaN               NaN  
...                   ...               ...  
1042163               NaN               NaN  
1042164               NaN               NaN  
1042165               NaN               NaN  
1042166               NaN               NaN  
1042167               NaN               NaN  

[1042168 rows x 11 columns]

### ข้อมูลโควิด ตั้งแต่วันที่ 26 ตุลาคม 2021 ถึง 27 กุมภาพันธุ์ 2022

In [70]:
data_file_path3 = os.path.join(path,'confirmed-cases-since-271064.csv')
data_covid3 = pd.read_csv(data_file_path3)

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2882: DtypeWarning: Columns (4) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [71]:
data_covid3

No. announce_date Notified date   sex   age Unit nationality  \
0        1859158    26/10/2021    25/10/2021   ชาย  20.0   ปี    Thailand   
1        1859159    26/10/2021    25/10/2021  หญิง  42.0   ปี    Thailand   
2        1859160    26/10/2021    25/10/2021  หญิง  33.0   ปี    Thailand   
3        1859161    26/10/2021    25/10/2021  หญิง  52.0   ปี    Thailand   
4        1859162    26/10/2021    25/10/2021  หญิง  84.0   ปี    Thailand   
...          ...           ...           ...   ...   ...  ...         ...   
1010454  2869612     27/2/2022     26/2/2022   ชาย   NaN  NaN         NaN   
1010455  2869613     27/2/2022     26/2/2022   ชาย   NaN  NaN         NaN   
1010456  2869614     27/2/2022     26/2/2022   ชาย   NaN  NaN         NaN   
1010457  2869615     27/2/2022     26/2/2022   ชาย   NaN  NaN         NaN   
1010458  2869616     27/2/2022     26/2/2022   ชาย   NaN  NaN         NaN   

        province_of_isolation                                           risk  \
0                       ลำปาง                                          อื่นๆ   
1                       ลำปาง                                          อื่นๆ   
2                       ลำปาง  ไปสถานที่ชุมชน เช่น ตลาดนัด สถานที่ท่องเที่ยว   
3                       ลำปาง  ไปสถานที่ชุมชน เช่น ตลาดนัด สถานที่ท่องเที่ยว   
4                       ลำปาง    สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
...                       ...                                            ...   
1010454         กรุงเทพมหานคร                               ทัณฑสถาน/เรือนจำ   
1010455         กรุงเทพมหานคร                               ทัณฑสถาน/เรือนจำ   
1010456               ปัตตานี                               ทัณฑสถาน/เรือนจำ   
1010457               ปัตตานี                               ทัณฑสถาน/เรือนจำ   
1010458               ปัตตานี                               ทัณฑสถาน/เรือนจำ   

        province_of_onset district_of_onset  
0                   ลำปาง        เมืองลำปาง  
1                   ลำปาง        เมืองลำปาง  
2                   ลำปาง          ห้างฉัตร  
3                   ลำปาง          ห้างฉัตร  
4                   ลำปาง        เมืองลำปาง  
...                   ...               ...  
1010454               NaN               NaN  
1010455               NaN               NaN  
1010456               NaN               NaN  
1010457               NaN               NaN  
1010458               NaN               NaN  

[1010459 rows x 11 columns]

### ข้อมูลโควิด ตั้งแต่วันที่ 28 กุมภาพันธุ์ 2022 ถึง 12 มีนาคม 2022

In [72]:
data_file_path4 = os.path.join(path,'confirmed-cases-since-280265.csv')
data_covid4 = pd.read_csv(data_file_path4)

In [73]:
data_covid4

No. announce_date Notified date   sex   age Unit nationality  \
0       2869617     28/2/2022     27/2/2022   ชาย  41.0   ปี    Thailand   
1       2869618     28/2/2022     27/2/2022  หญิง  45.0   ปี    Thailand   
2       2869619     28/2/2022     27/2/2022  หญิง  40.0   ปี    Thailand   
3       2869620     28/2/2022     27/2/2022   ชาย  44.0   ปี    Thailand   
4       2869621     28/2/2022     27/2/2022   ชาย  11.0   ปี    Thailand   
...         ...           ...           ...   ...   ...  ...         ...   
291620  3161237     12/3/2022     11/3/2022  หญิง   NaN  NaN         NaN   
291621  3161238     12/3/2022     11/3/2022   ชาย   NaN  NaN   Cambodian   
291622  3161239     12/3/2022     11/3/2022  หญิง   NaN  NaN         NaN   
291623  3161240     12/3/2022     11/3/2022  หญิง   NaN  NaN         NaN   
291624  3161241     12/3/2022     11/3/2022   NaN   NaN  NaN         NaN   

       province_of_isolation                                         risk  \
0                      พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
1                      พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
2                      พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
3                      พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
4                      พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
...                      ...                                          ...   
291620         กรุงเทพมหานคร                             ทัณฑสถาน/เรือนจำ   
291621         กรุงเทพมหานคร                             ทัณฑสถาน/เรือนจำ   
291622         กรุงเทพมหานคร                             ทัณฑสถาน/เรือนจำ   
291623         กรุงเทพมหานคร                             ทัณฑสถาน/เรือนจำ   
291624             เชียงใหม่                             ทัณฑสถาน/เรือนจำ   

       province_of_onset district_of_onset  
0                  พะเยา        เมืองพะเยา  
1                  พะเยา        เมืองพะเยา  
2                  พะเยา        เมืองพะเยา  
3                  พะเยา        เมืองพะเยา  
4                  พะเยา        เมืองพะเยา  
...                  ...               ...  
291620               NaN               NaN  
291621               NaN               NaN  
291622               NaN               NaN  
291623               NaN               NaN  
291624               NaN               NaN  

[291625 rows x 11 columns]

### ข้อมูลภูมิภาค

สร้างตารางภูมิภาค ที่มา : https://th.wikipedia.org/wiki/%E0%B8%A0%E0%B8%B9%E0%B8%A1%E0%B8%B4%E0%B8%A0%E0%B8%B2%E0%B8%84%E0%B8%82%E0%B8%AD%E0%B8%87%E0%B8%9B%E0%B8%A3%E0%B8%B0%E0%B9%80%E0%B8%97%E0%B8%A8%E0%B9%84%E0%B8%97%E0%B8%A2

ใช้ข้อมูลภูมิภาคแบ่งตามการเมือง ซึ่งประกอบไปด้วย 4 ภูมิภาค คือ ภาคเหนือ ภาคตะวันออกเฉียงเหนือ ภาคกลางและภาคใต้ 

***เพื่อต้องการรู้ว่าในแต่ละภูมิภาคมีจำนวนผู้ติดเชื้อเป็นอย่างไร ภาคใดมีผู้ติดเชื้อมากที่สุด***

In [74]:
# ใช้วิธี List(Row Oriented)
data =[['อำนาจเจริญ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['บึงกาฬ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['บุรีรัมย์','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['ชัยภูมิ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['กาฬสินธุ์','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['ขอนแก่น','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['เลย','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['มหาสารคาม','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['มุกดาหาร','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['นครพนม','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['นครราชสีมา','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['หนองบัวลำภู','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['หนองคาย','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['ร้อยเอ็ด','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['สกลนคร','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['ศรีสะเกษ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['สุรินทร์','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['อุบลราชธานี','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['อุดรธานี','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['ยโสธร','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ'],
            ['เชียงใหม่','เหนือ','เหนือ','เหนือ','เหนือ'],
            ['เชียงราย','เหนือ','เหนือ','เหนือ','เหนือ'],
            ['ลำปาง','เหนือ','เหนือ','เหนือ','เหนือ'],
            ['ลำพูน','เหนือ','เหนือ','เหนือ','เหนือ'],
            ['แม่ฮ่องสอน','เหนือ','เหนือ','เหนือ','เหนือ'],
            ['น่าน','เหนือ','เหนือ','เหนือ','เหนือ'],
            ['พะเยา','เหนือ','เหนือ','เหนือ','เหนือ'],
            ['แพร่','เหนือ','เหนือ','เหนือ','เหนือ'],
            ['อุตรดิตถ์','เหนือ','เหนือ','เหนือ','เหนือ'],
            ['ตาก','ตะวันตก','เหนือ','เหนือ','เหนือ'],
            ['สุโขทัย','กลาง','เหนือ','เหนือ','เหนือ'],
            ['พิษณุโลก','กลาง','เหนือ','เหนือ','เหนือ'],
            ['พิจิตร','กลาง','เหนือ','เหนือ','เหนือ'],
            ['กำแพงเพชร','กลาง','เหนือ','เหนือ','เหนือ'],
            ['เพชรบูรณ์','กลาง','เหนือ','เหนือ','เหนือ'],
            ['นครสวรรค์','กลาง','เหนือ','กลาง','เหนือ'],
            ['อุทัยธานี','กลาง','เหนือ','กลาง','เหนือ'],
            ['อ่างทอง','กลาง','กลาง','กลาง','กลาง'],
            ['ชัยนาท','กลาง','กลาง','กลาง','กลาง'],
            ['พระนครศรีอยุธยา','กลาง','กลาง','กลาง','กลาง'],
            ['กรุงเทพมหานคร','กลาง','กลาง','กลาง','กลาง'],
            ['ลพบุรี','กลาง','กลาง','กลาง','กลาง'],
            ['นครปฐม','กลาง','กลาง','กลาง','กลาง'],
            ['นนทบุรี','กลาง','กลาง','กลาง','กลาง'],
            ['ปทุมธานี','กลาง','กลาง','กลาง','กลาง'],
            ['สมุทรปราการ','กลาง','กลาง','กลาง','กลาง'],
            ['สมทรสาคร','กลาง','กลาง','กลาง','กลาง'],
            ['สมุทรสงคราม','กลาง','กลาง','กลาง','กลาง'],
            ['สระบุรี','กลาง','กลาง','กลาง','กลาง'],
            ['สิงห์บุรี','กลาง','กลาง','กลาง','กลาง'],
            ['สุพรรณบุรี','กลาง','กลาง','กลาง','กลาง'],
            ['นครนายก','กลาง','กลาง','ตะวันออก','กลาง'],
            ['ฉะเชิงเทรา','ตะวันออก','กลาง','ตะวันออก','ตะวันออก'],
            ['จันทบุรี','ตะวันออก','กลาง','ตะวันออก','ตะวันออก'],
            ['ชลบุรี','ตะวันออก','กลาง','ตะวันออก','ตะวันออก'],
            ['ปราจีนบุรี','ตะวันออก','กลาง','ตะวันออก','ตะวันออก'],
            ['ระยอง','ตะวันออก','กลาง','ตะวันออก','ตะวันออก'],
            ['สระแก้ว','ตะวันออก','กลาง','ตะวันออก','ตะวันออก'],
            ['ตราด','ตะวันออก','กลาง','ตะวันออก','ตะวันออก'],
            ['กาญจนบุรี','ตะวันตก','กลาง','กลาง','กลาง'],
            ['ราชบุรี','ตะวันตก','กลาง','กลาง','กลาง'],
            ['เพชรบุรี','ตะวันตก','กลาง','ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย','กลาง'],
            ['ประจวบคีรีขันธ์','ตะวันตก','กลาง','ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย','กลาง'],
            ['ชุมพร','ใต้','ใต้','ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย','ใต้'],
            ['นครศรีธรรมราช','ใต้','ใต้','ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย','ใต้'],
            ['นราธิวาส','ใต้','ใต้','ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย','ใต้'],
            ['ปัตตานี','ใต้','ใต้','ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย','ใต้'],
            ['พัทลุง','ใต้','ใต้','ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย','ใต้'],
            ['สงขลา','ใต้','ใต้','ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย','ใต้'],
            ['สุราษฎร์ธานี','ใต้','ใต้','ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย','ใต้'],
            ['ยะลา','ใต้','ใต้','ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย','ใต้'],
            ['กระบี่','ใต้','ใต้','ใต้ฝั่งตะวันตก/ใต้ฝั่งอันดามัน','ใต้'],
            ['พังงา','ใต้','ใต้','ใต้ฝั่งตะวันตก/ใต้ฝั่งอันดามัน','ใต้'],
            ['ภูเก็ต','ใต้','ใต้','ใต้ฝั่งตะวันตก/ใต้ฝั่งอันดามัน','ใต้'],
            ['ระนอง','ใต้','ใต้','ใต้ฝั่งตะวันตก/ใต้ฝั่งอันดามัน','ใต้'],
            ['สตูล','ใต้','ใต้','ใต้ฝั่งตะวันตก/ใต้ฝั่งอันดามัน','ใต้'],
            ['ตรัง','ใต้','ใต้','ใต้ฝั่งตะวันตก/ใต้ฝั่งอันดามัน','ใต้']]
headers = ['จังหวัด','ภูมิศาสตร์','การเมือง','อุตุนิยมวิทยา','การท่องเที่ยว']

In [75]:
data_region= pd.DataFrame.from_records(data,columns=headers) # สร้าง dataframe
data_region

จังหวัด          ภูมิศาสตร์            การเมือง  \
0   อำนาจเจริญ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ   
1       บึงกาฬ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ   
2    บุรีรัมย์  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ   
3      ชัยภูมิ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ   
4    กาฬสินธุ์  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ   
..         ...                 ...                 ...   
72       พังงา                 ใต้                 ใต้   
73      ภูเก็ต                 ใต้                 ใต้   
74       ระนอง                 ใต้                 ใต้   
75        สตูล                 ใต้                 ใต้   
76        ตรัง                 ใต้                 ใต้   

                     อุตุนิยมวิทยา       การท่องเที่ยว  
0               ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ  
1               ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ  
2               ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ  
3               ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ  
4               ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ  
..                             ...                 ...  
72  ใต้ฝั่งตะวันตก/ใต้ฝั่งอันดามัน                 ใต้  
73  ใต้ฝั่งตะวันตก/ใต้ฝั่งอันดามัน                 ใต้  
74  ใต้ฝั่งตะวันตก/ใต้ฝั่งอันดามัน                 ใต้  
75  ใต้ฝั่งตะวันตก/ใต้ฝั่งอันดามัน                 ใต้  
76  ใต้ฝั่งตะวันตก/ใต้ฝั่งอันดามัน                 ใต้  

[77 rows x 5 columns]

### ข้อมูลเขตสุขภาพ ที่มา : https://www.mhc2.go.th/newweb/information?fbclid=IwAR3xS9z4R1Gv-iNL0RkIqT_vUWdLK0LHoF8pojdt0VW-dAr0HH1ImbIA-ug#:~:text=1.%20%E0%B8%AA%E0%B9%88%E0%B8%A7%E0%B8%99%E0%B8%81%E0%B8%A5%E0%B8%B2%E0%B8%87%20%E0%B9%84%E0%B8%94%E0%B9%89%E0%B9%81%E0%B8%81%E0%B9%88%20%E0%B8%81%E0%B8%A3%E0%B8%B8%E0%B8%87%E0%B9%80%E0%B8%97%E0%B8%9E%E0%B8%A1%E0%B8%AB%E0%B8%B2%E0%B8%99%E0%B8%84%E0%B8%A3,%E0%B8%81%E0%B8%B3%E0%B9%81%E0%B8%9E%E0%B8%87%E0%B9%80%E0%B8%9E%E0%B8%8A%E0%B8%A3%20%E0%B8%9E%E0%B8%B4%E0%B8%88%E0%B8%B4%E0%B8%95%E0%B8%A3%20%E0%B8%99%E0%B8%84%E0%B8%A3%E0%B8%AA%E0%B8%A7%E0%B8%A3%E0%B8%A3%E0%B8%84%E0%B9%8C%20%E0%B8%8A%E0%B8%B1%E0%B8%A2%E0%B8%99%E0%B8%B2%E0%B8%97%20%E0%B8%AD%E0%B8%B8%E0%B8%97%E0%B8%B1%E0%B8%A2%E0%B8%98%E0%B8%B2%E0%B8%99%E0%B8%B5

ข้อมูลเขตสุขภาพนั้น แบ่งเป็นเขตสุขภาพส่วนกลาง(กรุงเทพมหานคร)และเป็นทั้งหมด 12 เขต คือ เขตที่1 - 12 ซึ่งเป็นระบบบริการสุขภาพ สร้างเสริมสุขภาพ ควบคุมป้องกันโรค และระบบยาและการคุ้มครองผู้บริโภคของแต่ละโรงพยาบาลในจังหวัดตามเขตสุขภาพนั้นๆ โดยที่ประชาชนสามารถเข้าถึงการบริการได้สะดวกและรวดเร็ว 

***เพื่อต้องการรู้ว่าแต่ละเขตสุขภาพนั้นต้องรับรักษาผู้ติดเชื้อมากน้อยเพียงใด เขตสุขภาพใดมีผู้ติดเชื้อมากที่สุด***

วิธี Dictionary(column oriented) สร้างตัวเเปร data_mhc ขึ้นมา เเล้วกำหนดคอลัมน์ ได้เเก่ จังหวัด, เขตสุขภาพ

In [76]:
data_mhc = {'จังหวัด':['กรุงเทพมหานคร', 'เชียงใหม่','เชียงราย','แพร่', 'น่าน', 'พะเยา', 'ลำปาง', 'ลำพูน', 'แม่ฮ่องสอน', 'ตาก', 'เพชรบูรณ์', 'พิษณุโลก', 'อุดรดิตถ์',
                       'สุโขทัย', 'กำแพงเพชร', 'พิจิตร', 'นครสวรรค์', 'ชัยนาท', 'อุทัยธานี', 'สระบุรี', 'นนทบุรี', 'ลพบุรี', 'อ่างทอง', 'นครนายก', 'สิงห์บุรี',
                       'พระนครศรีอยุธยา', 'ปทุมธานี', 'เพชรบุรี', 'สมุทรสาคร', 'สมุทรสงคราม', 'ประจวบคีรีขันธ์', 'สุพรรณบุรี', 'นครปฐม', 'ราชบุรี', 'กาญจนบุรี',
                       'กาญจนบุรี', 'สระแก้ว', 'ปราจีนบุรี', 'ฉะเชิงเทรา', 'สมุทรปราการ', 'ชลบุรี', 'จันทบุรี', 'ระยอง', 'ตราด', 'กาฬสินธุ์', 'ขอนแก่น', 'มหาสารคาม',
                       'ร้อยเอ็ด', 'อุดรธานี', 'สกลนคร', 'นครพนม', 'เลย', 'หนองคาย', 'หนองบัวลำภู', 'บึงกาฬ', 'ชัยภูมิ', 'นครราชสีมา', 'บุรีรัมย์', 'สุรินทร์',
                       'อุบลราชธานี', 'ศรีสะเกษ', 'มุกดาหาร', 'อำนาจเจริญ', 'นครศรีธรรมราช', 'สุราษฎร์ธานี', 'ภูเก็ต', 'กระบี่', 'พังงา', 'ระยอง', 'ชุมพร',
                       'สงขลา', 'สตูล', 'ตรัง', 'พัทลุง', 'ปัตตานี', 'ยะลา', 'นราธิวาส'],
            'เขตสุขภาพ':['เขตสุขภาพส่วนกลาง', 'เขตสุขภาพที่ 1', 'เขตสุขภาพที่ 1', 'เขตสุขภาพที่ 1', 'เขตสุขภาพที่ 1', 'เขตสุขภาพที่ 1', 'เขตสุขภาพที่ 1', 'เขตสุขภาพที่ 1', 'เขตสุขภาพที่ 1', 'เขตสุขภาพที่ 1',
                         'เขตสุขภาพที่ 2','เขตสุขภาพที่ 2','เขตสุขภาพที่ 2','เขตสุขภาพที่ 2','เขตสุขภาพที่ 2','เขตสุขภาพที่ 3','เขตสุขภาพที่ 3','เขตสุขภาพที่ 3','เขตสุขภาพที่ 3','เขตสุขภาพที่ 3',
                         'เขตสุขภาพที่ 4','เขตสุขภาพที่ 4','เขตสุขภาพที่ 4','เขตสุขภาพที่ 4','เขตสุขภาพที่ 4','เขตสุขภาพที่ 4','เขตสุขภาพที่ 4','เขตสุขภาพที่ 4','เขตสุขภาพที่ 5','เขตสุขภาพที่ 5',
                         'เขตสุขภาพที่ 5','เขตสุขภาพที่ 5','เขตสุขภาพที่ 5','เขตสุขภาพที่ 5','เขตสุขภาพที่ 5','เขตสุขภาพที่ 5','เขตสุขภาพที่ 6','เขตสุขภาพที่ 6','เขตสุขภาพที่ 6','เขตสุขภาพที่ 6',
                         'เขตสุขภาพที่ 6','เขตสุขภาพที่ 6','เขตสุขภาพที่ 6','เขตสุขภาพที่ 6','เขตสุขภาพที่ 7','เขตสุขภาพที่ 7','เขตสุขภาพที่ 7','เขตสุขภาพที่ 7','เขตสุขภาพที่ 8','เขตสุขภาพที่ 8',
                         'เขตสุขภาพที่ 8','เขตสุขภาพที่ 8','เขตสุขภาพที่ 8','เขตสุขภาพที่ 8','เขตสุขภาพที่ 8','เขตสุขภาพที่ 9','เขตสุขภาพที่ 9','เขตสุขภาพที่ 9','เขตสุขภาพที่ 9','เขตสุขภาพที่ 10',
                         'เขตสุขภาพที่ 10','เขตสุขภาพที่ 10','เขตสุขภาพที่ 10','เขตสุขภาพที่ 11','เขตสุขภาพที่ 11','เขตสุขภาพที่ 11','เขตสุขภาพที่ 11','เขตสุขภาพที่ 11','เขตสุขภาพที่ 11','เขตสุขภาพที่ 11',
                         'เขตสุขภาพที่ 12','เขตสุขภาพที่ 12','เขตสุขภาพที่ 12','เขตสุขภาพที่ 12','เขตสุขภาพที่ 12','เขตสุขภาพที่ 12','เขตสุขภาพที่ 12']}

In [77]:
data_mentalhealth= pd.DataFrame.from_dict(data_mhc)  # สร้างตาราง pandas จาก data_mhc 
data_mentalhealth #เเสดงข้อมูลในตารางทั้งหมด

จังหวัด          เขตสุขภาพ
0   กรุงเทพมหานคร  เขตสุขภาพส่วนกลาง
1       เชียงใหม่     เขตสุขภาพที่ 1
2        เชียงราย     เขตสุขภาพที่ 1
3            แพร่     เขตสุขภาพที่ 1
4            น่าน     เขตสุขภาพที่ 1
..            ...                ...
72           ตรัง    เขตสุขภาพที่ 12
73         พัทลุง    เขตสุขภาพที่ 12
74        ปัตตานี    เขตสุขภาพที่ 12
75           ยะลา    เขตสุขภาพที่ 12
76       นราธิวาส    เขตสุขภาพที่ 12

[77 rows x 2 columns]

***รวมข้อมูล covid ให้อยู่ในตารางเดียวกัน***

แต่เนื่องจากว่าในตาราง data_covid2 นั้นมีตารางที่เกินมาจากตารางอื่น ๆ จึงทำการตัดตารางที่เกินมาทิ้งไปก่อน

In [78]:
data_covid = pd.concat([data_covid1,data_covid2,data_covid3,data_covid4])
data_covid

No. announce_date Notified date   sex   age Unit nationality  \
0             1     12/1/2020           NaN  หญิง  61.0   ปี       China   
1             2     17/1/2020           NaN  หญิง  74.0   ปี       China   
2             3     22/1/2020           NaN  หญิง  73.0   ปี    Thailand   
3             4     22/1/2020           NaN   ชาย  68.0   ปี       China   
4             5     24/1/2020           NaN  หญิง  66.0   ปี       China   
...         ...           ...           ...   ...   ...  ...         ...   
291620  3161237     12/3/2022     11/3/2022  หญิง   NaN  NaN         NaN   
291621  3161238     12/3/2022     11/3/2022   ชาย   NaN  NaN   Cambodian   
291622  3161239     12/3/2022     11/3/2022  หญิง   NaN  NaN         NaN   
291623  3161240     12/3/2022     11/3/2022  หญิง   NaN  NaN         NaN   
291624  3161241     12/3/2022     11/3/2022   NaN   NaN  NaN         NaN   

       province_of_isolation                              risk  \
0              กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ   
1              กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ   
2                     นครปฐม  คนต่างชาติเดินทางมาจากต่างประเทศ   
3              กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ   
4                    นนทบุรี  คนต่างชาติเดินทางมาจากต่างประเทศ   
...                      ...                               ...   
291620         กรุงเทพมหานคร                  ทัณฑสถาน/เรือนจำ   
291621         กรุงเทพมหานคร                  ทัณฑสถาน/เรือนจำ   
291622         กรุงเทพมหานคร                  ทัณฑสถาน/เรือนจำ   
291623         กรุงเทพมหานคร                  ทัณฑสถาน/เรือนจำ   
291624             เชียงใหม่                  ทัณฑสถาน/เรือนจำ   

       province_of_onset district_of_onset  
0          กรุงเทพมหานคร               NaN  
1          กรุงเทพมหานคร               NaN  
2                 นครปฐม             เมือง  
3          กรุงเทพมหานคร               NaN  
4          กรุงเทพมหานคร               NaN  
...                  ...               ...  
291620               NaN               NaN  
291621               NaN               NaN  
291622               NaN               NaN  
291623               NaN               NaN  
291624               NaN               NaN  

[3184023 rows x 11 columns]

## จัดการค่า Missing

In [79]:
data_covid.isnull().any() # ดูว่ามีค่า missing หรือไม่

No.                      False
announce_date            False
Notified date             True
sex                       True
age                       True
Unit                      True
nationality               True
province_of_isolation     True
risk                      True
province_of_onset         True
district_of_onset         True
dtype: bool

### จัดการค่า Missing ของ sex โดยจะแทนโดยค่า Mode

In [80]:
set(data_covid['sex']) # ดูว่า sex เก็บค่าอะไรไว้บ้าง

{nan,
 '\xa0หญิง',
 'ช',
 'ชาย',
 'ญ',
 'นาย',
 'ร.ต.ท.',
 'หญง',
 'หญิง',
 'หญิิง',
 'หณิง',
 'หห'}

In [81]:
data_covid['sex'].mode() # หาค่า mode ของ sex

0    หญิง
dtype: object

In [82]:
data_covid[['sex']] = data_covid[['sex']].fillna(value={'sex':'หญิง'}) # แทนค่า mode ในค่า missing
data_covid[['sex']]

sex
0       หญิง
1       หญิง
2       หญิง
3        ชาย
4       หญิง
...      ...
291620  หญิง
291621   ชาย
291622  หญิง
291623  หญิง
291624  หญิง

[3184023 rows x 1 columns]

In [83]:
# แก้ข้อมูลที่ผิดให้อยู่ในรูปแบบเดียวกัน
data_covid['sex'].replace({'\xa0หญิง':'หญิง',
                            'ช':'ชาย',
                            'ญ':'หญิง',
                            'นาย':'ชาย',
                            'ร.ต.ท.':'ชาย',
                            'หญง':'หญิง',
                            'หญิิง':'หญิง',
                            'หณิง':'หญิง',
                            'หห':'หญิง'}, inplace=True)

In [84]:
set(data_covid['sex'])

{'ชาย', 'หญิง'}

In [85]:
data_covid['sex'].isnull().any()

False

### จัดการค่า Missing ของ risk โดยจะแทนโดย ไม่ระบุ

In [86]:
data_covid[['risk']] = data_covid[['risk']].fillna(value={'risk':'ไม่ระบุ'}) # แทนค่า ไม่ระบุ ในค่า missing
data_covid[['risk']]

risk
0       คนต่างชาติเดินทางมาจากต่างประเทศ
1       คนต่างชาติเดินทางมาจากต่างประเทศ
2       คนต่างชาติเดินทางมาจากต่างประเทศ
3       คนต่างชาติเดินทางมาจากต่างประเทศ
4       คนต่างชาติเดินทางมาจากต่างประเทศ
...                                  ...
291620                  ทัณฑสถาน/เรือนจำ
291621                  ทัณฑสถาน/เรือนจำ
291622                  ทัณฑสถาน/เรือนจำ
291623                  ทัณฑสถาน/เรือนจำ
291624                  ทัณฑสถาน/เรือนจำ

[3184023 rows x 1 columns]

In [87]:
set(data_covid['risk'])

{' Samui plus',
 '10.อื่นๆ',
 '2.สัมผัสผู้ติดเชื้อ',
 '8.สำรวจกลุ่มเสี่ยง (survey)',
 '9.ขอตรวจหาเชื้อเอง',
 'ACF รองเมือง',
 'ACF สนามกีฬาธูปะเตมีย์',
 'ACF สนามกีฬาไทย-ญี่ปุ่น',
 'ACF สีลม',
 'ACF เคหะหลักสี่',
 'BKK Sandbox',
 'Chonburi Sandbox',
 'Cluster Icon siam',
 "Cluster Memory 90's กรุงเทพมหานคร",
 'Cluster New Jazz กรุงเทพมหานคร',
 'Cluster The Lounge Salaya',
 'Cluster คลองเตย',
 'Cluster งานมอเตอร์ โชว์',
 'Cluster จนท. สนามบินสุวรรณภูมิ',
 'Cluster จันทบุรี',
 'Cluster จันทบุรี (ชาวกินี )',
 'Cluster จันทบุรี (ชาวกินี)',
 'Cluster ชลบุรี',
 'Cluster ชลบุรี บริษัทไดกิ้น',
 'Cluster ชลบุรี โรงเบียร์ 90',
 'Cluster ตราด',
 'Cluster ตราด ',
 'Cluster ตลาดนนทบุรี',
 'Cluster ตลาดบางพลี',
 'Cluster ตลาดบางแค',
 'Cluster ตลาดพรพัฒน์',
 'Cluster ตลาดสี่มุมเมือง',
 'Cluster บริษัทศรีสวัสดิ์',
 'Cluster บางแค',
 'Cluster บ่อนพัทยา/ชลบุรี',
 'Cluster บ่อนเทพารักษ์',
 'Cluster บ่อนไก่อ่างทอง',
 'Cluster ปากช่อง',
 'Cluster ผับ Thonglor',
 'Cluster มหาสารคาม',
 'Cluster ระยอง',
 'Clu

In [88]:
# แก้ข้อมูลที่ผิดให้อยู่ในรูปแบบเดียวกัน
data_covid['risk'].replace({' Samui plus':'Samui Plus',
                            '10.อื่นๆ':'อื่น ๆ',
                            '2.สัมผัสผู้ติดเชื้อ':'สัมผัสผู้ติดเชื้อ',
                            '8.สำรวจกลุ่มเสี่ยง (survey)':'กลุ่มเสี่ยง',
                            '9.ขอตรวจหาเชื้อเอง':'ขอตรวจหาเชื้อเอง',
                            'Cluster จันทบุรี (ชาวกินี )':'Cluster จันทบุรี (ชาวกินี)',
                            'Cluster ตราด ':'Cluster ตราด',
                            'Karbi Sandbox':'Krabi Sandbox',
                            'Krabi Sandbox ':'Krabi Sandbox',
                            'Krabi sandbox ':'Krabi Sandbox',
                            'Phang nga Sandbox':'Phang Nga Sandbox',
                            'Phangnga Sandbox':'Phang Nga Sandbox',
                            'Phare Sandbox':'Phrae Sandbox',
                            'Phunga Sandbox':'Phung Nga Sandbox',
                            'SHUI':'อื่นๆ',
                            'Samui plus':'Samui Plus',
                            'cluster ชลบุรี\n(อยู่ระหว่างการสอบสวน)':'cluster ชลบุรี',
                            'prachuap Sandbox':'Prachuap Sandbox',
                            'กลับมาจาก จ.กรุงเทพมหานคร.':'ผู้เดินทางมาจากพื้นที่เสี่ยง',
                            'คนไทยมาจากพื้นที่เสี่ยง':'ผู้เดินทางมาจากพื้นที่เสี่ยง',
                            'คนต่างชาติเดินทางมาจากต่างประเทศ':'ผู้ที่เดินทางมาจากต่างประเทศ',
                            'คนไทยเดินทางกลับจากต่างประเทศ':'ผู้ที่เดินทางมาจากต่างประเทศ',
                            'คนไทยเดินทางมาจากต่างประเทศ':'ผู้ที่เดินทางมาจากต่างประเทศ',
                            'ติดเชื้อในพื้นที่ (สัมผัส ผป.ยืนยัน)':'พื้นที่เสียง',
                            'บุคลากรทางการเเพทย์':'บุคลากรด้านการแพทย์และสาธารณสุข',
                            'ปอดอักเสบ Pneumonia':'ปอดอักเสบ (Pneumonia)',
                            'ปอดอักเสบ(Pneumonia)':'ปอดอักเสบ (Pneumonia)',
                            'ผับหัวหิน':'สถานบันเทิง',
                            'ผู้ติดเชื้อในประเทศ':'สัมผัสผู้ป่วยยืนยัน',
                            'ผู้ที่เดินทางมาจากต่างประเทศ และเข้า AOQ':'ผู้ที่เดินทางมาจากต่างประเทศ',
                            'ผู้ที่เดินทางมาจากต่างประเทศ และเข้า AQ':'ผู้ที่เดินทางมาจากต่างประเทศ',
                            'ผู้ที่เดินทางมาจากต่างประเทศ และเข้า ASQ/ALQ':'ผู้ที่เดินทางมาจากต่างประเทศ',
                            'ผู้ที่เดินทางมาจากต่างประเทศ และเข้า HQ/AHQ':'ผู้ที่เดินทางมาจากต่างประเทศ',
                            'ผู้ที่เดินทางมาจากต่างประเทศ และเข้า OQ':'ผู้ที่เดินทางมาจากต่างประเทศ',
                            'ผู้ป่วย PUI':'สัมผัสผู้ป่วยยืนยัน',
                            'ผู้ป่วย กรุงเทพมหานคร':'สัมผัสผู้ป่วยยืนยัน',
                            'ผู้เดินทางมาจากพื้นที่เสี่ยง กรุงเทพมหานคร':'ผู้เดินทางมาจากพื้นที่เสี่ยง',
                            'มาจากพื้นที่เสียง จ.ปทุมธานี':'ผู้เดินทางมาจากพื้นที่เสี่ยง',
                            'มาจากพื้นที่เสี่ยง':'ผู้เดินทางมาจากพื้นที่เสี่ยง',
                            'มาจากพื้นที่เสี่ยง จ.ปทุมธานี':'ผู้เดินทางมาจากพื้นที่เสี่ยง',
                            'มีอาการ มาจาก กรุงเทพมหานคร. พนักงานร้านอาหาร':'ผู้เดินทางมาจากพื้นที่เสี่ยง',
                            'ระบุไม่ได้':'ไม่ระบุ',
                            'ระหว่างการสอบสวนโรค':'อยู่ระหว่างการสอบสวน',
                            'สัมผัสผู้ติดเชื้อยืนยัน\n  อยู่ระหว่างสอบสวน':'สัมผัสผู้ป่วยยืนยัน',
                            'สัมผัสผู้ติดเชื้อยืนยัน\nอยู่ระหว่างสอบสวน':'สัมผัสผู้ป่วยยืนยัน',
                            'สัมผัสผู้ป่วยที่ติดโควิด':'สัมผัสผู้ป่วยยืนยัน',
                            'สัมผัสผู้ป่วยยืนยัน \n  อยู่ระหว่างสอบสวน':'สัมผัสผู้ป่วยยืนยัน',
                            'สัมผัสผู้ป่วยยืนยัน (ยังไม่สามารถระบุได้)':'สัมผัสผู้ป่วยยืนยัน',
                            'สัมผัสผู้ป่วยยืนยัน ภายในครอบครัว/ชุมชน/เพื่อน':'สัมผัสผู้ป่วยยืนยัน',
                            'สัมผัสผู้ป่วยยืนยัน อยู่ระหว่างสอบสวน':'สัมผัสผู้ป่วยยืนยัน',
                            'สัมผัสผู้ป่วยยืนยันภายในชุมชน':'สัมผัสผู้ป่วยยืนยัน',
                            'สัมผัสผู้ป่วยยืนยันเพื่อนร่วมงาน':'สัมผัสผู้ป่วยยืนยัน',
                            'สัมผัสสามีสะอาด เขตบ้าน (รพ.มหาชัย 2)':'สัมผัสในสถานพยาบาล',
                            'สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้':'สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า',
                            'สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้ใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้':'สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า',
                            'สัมผัสใกล้ชิดผู้ป่วยยันยันก่อนหน้า':'สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า',
                            'อยู่ระหว่างการสอบสวน\n':'อยู่ระหว่างการสอบสวน',
                            'อยู่ระหว่างสอบสวน':'อยู่ระหว่างการสอบสวน',
                            'อาชีพเสี่ยง เช่น ทำงานในสถานที่แออัด หรือทำงานใกล้ชิดสัมผัสชาวต่างชาติ เป็นต้น':'อาชีพเสี่ยง',
                            'อิ่นๆ':'อื่น ๆ',
                            'อื่น':'อื่น ๆ',
                            'อื่นๆ':'อื่น ๆ',
                            'เดินทางจากพื้นที่เสี่ยง':'ผู้เดินทางมาจากพื้นที่เสี่ยง',
                            'เดินทางมาจากพื้นที่จังหวัดเสี่ยง':'ผู้เดินทางมาจากพื้นที่เสี่ยง',
                            'เดินทางมาจากพื้นที่มีการระบาดของโรค':'ผู้เดินทางมาจากพื้นที่เสี่ยง',
                            'เดินทางมาจากพื้นที่ระบาด':'ผู้เดินทางมาจากพื้นที่เสี่ยง',
                            'เดินทางมาจากพื้นที่เสี่ยง จ.กรุงเทพฯ จอมทอง':'ผู้เดินทางมาจากพื้นที่เสี่ยง',
                            'เดินทางมาจากพื้นที่เสี่ยงที่มีการระบาดของโรค':'ผู้เดินทางมาจากพื้นที่เสี่ยง',
                            'ในพื้นที่':'พื้นที่เสี่ยง',
                            'ไปยัง/มาจาก พื้นที่ระบาดกรุงเทพมหานคร':'ผู้เดินทางมาจากพื้นที่เสี่ยง',
                            'ไปยังพื้นที่ที่มีการระบาด':'ผู้เดินทางมาจากพื้นที่เสี่ยง',
                            'ไปสถานที่ชุมชน เช่น ตลาดนัด สถานที่ท่องเที่ยว':'ไปสถานที่ชุมชน',
                            'ไปสถานที่แออัด เช่น งานแฟร์ คอนเสิร์ต':'ไปสถานที่แออัด',
                            'ไปเที่ยวสถานบันเทิงในอุบลที่พบการระบาดของโรค Ubar':'สถานบันเทิง',
                            'ไปแหล่งชุมชน/สถานที่คนหนาแน่น':'ไปสถานที่ชุมชน'}, inplace=True)

In [89]:
data_covid['risk'].isnull().any()

False

### จัดการค่า Missing ของ province_of_onset โดยจะแทนค่า mode

In [90]:
data_covid['province_of_onset'].mode() # หาค่า mode ของ province_of_onset

0    กรุงเทพมหานคร
dtype: object

In [91]:
data_covid[['province_of_onset']] = data_covid[['province_of_onset']].fillna(value={'province_of_onset':'กรุงเทพมหานคร'}) # แทนค่า mode ในค่า missing
data_covid[['province_of_onset']]

province_of_onset
0          กรุงเทพมหานคร
1          กรุงเทพมหานคร
2                 นครปฐม
3          กรุงเทพมหานคร
4          กรุงเทพมหานคร
...                  ...
291620     กรุงเทพมหานคร
291621     กรุงเทพมหานคร
291622     กรุงเทพมหานคร
291623     กรุงเทพมหานคร
291624     กรุงเทพมหานคร

[3184023 rows x 1 columns]

In [92]:
set(data_covid['province_of_onset']) # ดูว่าเก็บค่าอะไรไว้บ้าง

{'\tประจวบคีรีขันธ์',
 'USA',
 '\xa0กาญจนบุรี',
 '\xa0ชลบุรี',
 '\xa0ตราด',
 '\xa0นครปฐม',
 '\xa0สมุทรสงคราม',
 '\xa0สมุทรสาคร',
 '\xa0อุทัยธานี',
 'กทม',
 'กทม.',
 'กบินทร์บุรี',
 'กระบี่',
 'กระบี่ี่',
 'กรุงเทพ',
 'กรุงเทพมหานคร',
 'กรุงเทพมหานคร.',
 'กรุงเทพมหานคร. / พระนครศรีอยุธยา',
 'กรุงเทพมหานคร. / ศรีสะเกษ',
 'กรุงเทพฯ',
 'กะทู้',
 'กัมพูชา',
 'กาญจนบุรี',
 'กาญจบุรี',
 'กาญนบุรี',
 'กาฬสินธิ์',
 'กาฬสินธุ์',
 'กาฬสินธุ์\xa0',
 'กาฬสินธ์',
 'กำเพงเพชร',
 'กำแพง',
 'กำแพงเพชร',
 'กำแพงเพชร\xa0',
 'กําแพงเพชร',
 'ขอนแก่น',
 'ขอนแก่น\xa0',
 'ขอนแก่่น',
 'ขัยนาท',
 'ขัยภูมิ',
 'ขานุวรลักษณ์บุรี',
 'คนไทยเดินทางกลับจากต่างประเทศ',
 'คลองเตย',
 'จ.กรุงเทพมหานคร',
 'จ.ภูเก็ต',
 'จ.ราชบุรี',
 'จ.สุราษฎร์ธานี',
 'จังหวัดสมุทรสาคร',
 'จันทบุรี',
 'ฉะเชองเทราร',
 'ฉะเชิงเทรา',
 'ชมพร',
 'ชลบุรี',
 'ชลบุรี\t',
 'ชลบุรี\xa0',
 'ชลบุุรี',
 'ชอนแก่น',
 'ชัยนาถ',
 'ชัยนาท',
 'ชัยบาดาล',
 'ชัยภูมิ',
 'ชัยภูมิ\xa0',
 'ชุมพร',
 'ชุุมพร',
 'ชุุุมพร',
 'ดอนมดแดง',
 'ดินแดง',
 'ตรัง',
 'ตรััง',
 '

In [93]:
data_covid['province_of_onset'].replace({'\tประจวบคีรีขันธ์':'ประจวบคีรีขันธ์',
                            'USA':'ไม่ระบุ',
                            '\xa0กาญจนบุรี':'กาญจนบุรี',
                            '\xa0ชลบุรี':'ชลบุรี',
                            '\xa0ตราด':'ตราด',
                            '\xa0นครปฐม':'นครปฐม',
                            '\xa0สมุทรสงคราม':'สมุทรสงคราม',
                            '\xa0อุทัยธานี':'อุทัยธานี',
                            'กทม':'กรุงเทพมหานคร',
                            'กทม.':'กรุงเทพมหานคร',
                            'กบินทร์บุรี':'ปราจีนบุรี',
                            'กระบี่ี่':'กระบี่',
                            'กรุงเทพ':'กรุงเทพมหานคร',
                            'กรุงเทพมหานคร.':'กรุงเทพมหานคร',
                            'กรุงเทพมหานคร. / พระนครศรีอยุธยา':'กรุงเทพมหานคร',
                            'กรุงเทพมหานคร. / ศรีสะเกษ':'กรุงเทพมหานคร',
                            'กรุงเทพฯ':'กรุงเทพมหานคร',
                            'กะทู้':'ภูเก็ต',
                            'กัมพูชา':'ไม่ระบุ',
                            'กาญนบุรี':'กาญจนบุรี',
                            'กาฬสินธิ์':'กาฬสินธุ์',
                            'กาฬสินธุ์\xa0':'กาฬสินธุ์',
                            'กาฬสินธ์':'กาฬสินธุ์',
                            'กำเพงเพชร':'กำแพงเพชร',
                            'กำแพง':'กำแพงเพชร',
                            'กำแพงเพชร\xa0':'กำแพงเพชร',
                            'ขอนแก่น\xa0':'ขอนแก่น',
                            'ขอนแก่่น':'ขอนแก่น',
                            'ขัยนาท':'ชัยนาท',
                            'ขัยภูมิ':'ชัยภูมิ',
                            'ขานุวรลักษณ์บุรี':'กำแพงเพชร',
                            'คนไทยเดินทางกลับจากต่างประเทศ':'กรุงเทพมหานคร',
                            'คลองเตย':'กรุงเทพมหานคร',
                            'จ.กรุงเทพมหานคร':'กรุงเทพมหานคร',
                            'จ.ภูเก็ต':'ภูเก็ต',
                            'จ.ราชบุรี':'ราชบุรี',
                            'จ.สุราษฎร์ธานี':'สุราษฎร์ธานี',
                            'จังหวัดสมุทรสาคร':'สมุทรสาคร',
                            'ฉะเชองเทราร':'ฉะเชิงเทรา',
                            'ชมพร':'ชุมพร',
                            'ชลบุรี\t':'ชลบุรี',
                            'ชลบุรี\xa0':'ชลบุรี',
                            'ชลบุุรี':'ชลบุรี',
                            'ชอนแก่น':'ขอนแก่น',
                            'ชัยนาถ':'ชัยนาท',
                            'ชัยบาดาล':'ลพบุรี',
                            'ชัยภูมิ\xa0':'ชัยภูมิ',
                            'ชุุมพร':'ชุมพร',
                            'ชุุุมพร':'ชุมพร',
                            'ดอนมดแดง':'อุบลราชธานี',
                            'ดินแดง':'กรุงเทพมหานคร',
                            'ตรััง':'ตรัง',
                            'ตราด\xa0':'ตราด',
                            'ตากใบ':'ตาก',
                            'ต่างจังหวัด':'ไม่ระบุ',
                            'ทุ่งเสลี่ยม':'สุโขทัย',
                            'ท้ายบ้าน':'สมุทรปราการ',
                            'ธารโต':'ยะลา',
                            'นคนปฐม':'นครปฐม',
                            'นคปฐม':'นครปฐม',
                            'นครนปฐม':'นครปฐม',
                            'นครปฐม\xa0':'นครปฐม',
                            'นครนาชสีมา':'นครราชสีมา',
                            'นครปบม':'นครพนม',
                            'นครพนม\xa0':'นครพนม',
                            'นครยายก':'นครนายก',
                            'นครราขสีมา':'นครราชสีมา',
                            'นครราชธานี':'นครราชสีมา',
                            'นครราชศรีมา':'นครราชสีมา',
                            'นครราชสีมา\xa0':'นครราชสีมา',
                            'นครราชสสีมา':'นครราชสีมา',
                            'นครศรธรรมราช':'นครศรีธรรมราช',
                            'นครศรี':'นครศรีธรรมราช',
                            'นครศรีธรมราช':'นครศรีธรรมราช',
                            'นครศรีธรรมราข':'นครศรีธรรมราช',
                            'นครศรีฯ':'นครศรีธรรมราช',
                            'นครศรีีธรรมราช':'นครศรีธรรมราช',
                            'นครศร๊ธรรมราช':'นครศรีธรรมราช',
                            'นครศร๊ะรรมราช':'นครศรีธรรมราช',
                            'นครสรรค์':'นครสวรรค์',
                            'นครสวรรค':'นครสวรรค์',
                            'นครสวรรค์\xa0':'นครสวรรค์',
                            'นครสีธรรมราช':'นครศรีธรรมราช',
                            'นครฯ':'นครศรีธรรมราช',
                            'นครีศรีธรรมราช':'นครศรีธรรมราช',
                            'นตรนายก':'นครนายก',
                            'นตรปฐม':'นครปฐม',
                            'นนทบุรี\xa0':'นนทบุรี',
                            'นนมบุรี':'นนทบุรี',
                            'นนทุรี':'นนทบุรี',
                            'นรศรีธรรมราช':'นครศรีธรรมราช',
                            'นราธิวาส\u200b':'นราธิวาส',
                            'นราธฺวาส':'นราธิวาส',
                            'นอกเขต':'กรุงเทพมหานคร',
                            'นาราธิวาส':'นราธิวาส',
                            'นาวัง':'หนองบัวลำภู',
                            'น่าน/ฉะเชิงเทรา':'น่าน',
                            'บางกะปิ':'กรุงเทพมหานคร',
                            'บางนา':'กรุงเทพมหานคร',
                            'บางปะอิน':'กรุงเทพมหานคร',
                            'บางละมุง':'กรุงเทพมหานคร',
                            'บางเพียง':'กรุงเทพมหานคร',
                            'บึงโขงหลง':'บึงกาฬ',
                            'บุรีรัมภ์':'บุรีรัมย์',
                            'บุรีรัมย์\xa0':'บุรีรัมย์',
                            'บุรีรัยม์':'บุรีรัมย์',
                            'บุรีรีมย์':'บุรีรัมย์',
                            'บ้านหลวง':'น่าน',
                            'ปจ':'ไม่ระบุ',
                            'ปทุทธานี':'ปทุมธานี',
                            'ปทุมฑานี':'ปทุมธานี',
                            'ปทุมธานี\xa0':'ปทุมธานี',
                            'ปทุมทานี':'ปทุมธานี',
                            'ปทุมธาน':'ปทุมธานี',
                            'ปทุมธานี.':'ปทุมธานี',
                            'ปทุมธานี\xa0':'ปทุมธานี',
                            'ปทุมธานีี':'ปทุมธานี',
                            'ปทุมาธานี':'ปทุมธานี',
                            'ปทุุมธานี':'ปทุมธานี',
                            'ปธุมธานี':'ปทุมธานี',
                            'ปมุมธานี':'ปทุมธานี',
                            'ประจวนคีรีขันธ็':'ประจวบคีรีขันธ์',
                            'ประจวบ':'ประจวบคีรีขันธ์',
                            'ประจวบคิรีขันธ์':'ประจวบคีรีขันธ์',
                            'ประจวบคีรขันธ์':'ประจวบคีรีขันธ์',
                            'ประจวบคีรขัันธ์':'ประจวบคีรีขันธ์',
                            'ประจวบคีรีขัน':'ประจวบคีรีขันธ์',
                            'ประจวบคีรีขันธื':'ประจวบคีรีขันธ์',
                            'ประจวบคีรีขันธฺ':'ประจวบคีรีขันธ์',
                            'ประจวบคีรีขันธ์\xa0':'ประจวบคีรีขันธ์',
                            'ประจวบคีรีขันธ์์':'ประจวบคีรีขันธ์',
                            'ประจวบคีรีขัันธ์':'ประจวบคีรีขันธ์',
                            'ประจวบคีีรีขันธ์':'ประจวบคีรีขันธ์',
                            'ประจวบฯ':'ประจวบคีรีขันธ์',
                            'ประทุมธานี':'ปทุมธานี',
                            'ประเทศกัมพูชา':'ไม่ระบุ',
                            'ปราจีบุรี':'ปราจีนบุรี',
                            'ปลวกแดง':'ระยอง',
                            'ปลายพระยา':'กระบี่',
                            'ปอยเปต':'ไม่ระบุ',
                            'ปัตตนรี':'ปัตตานี',
                            'ปัตตานนี':'ปัตตานี',
                            'ปัตตานีี':'ปัตตานี',
                            'ปัตาตานี':'ปัตตานี',
                            'ปัว':'ปัตตานี',
                            'ปััตตานี':'ปัตตานี',
                            'ปุทมธานี':'ปทุมธานี',
                            'ป่าโมก':'อ่างทอง',
                            'พม่า':'ไม่ระบุ',
                            'พรหมพิราม':'พิษณุโลก',
                            'พระนครศรียุธยา':'พระนครศรีอยุธยา',
                            'พระนครศรีอยุทธยา':'พระนครศรีอยุธยา',
                            'พระนครศรีอยุทยา':'พระนครศรีอยุธยา',
                            'พระนครศรีอยุธยา\xa0':'พระนครศรีอยุธยา',
                            'พระนครศรีอยุทยา':'พระนครศรีอยุธยา',
                            'พัทลุุง':'พัทลุง',
                            'พัังงา':'พังงา',
                            'พิษณุโลก\xa0':'พิษณุโลก',
                            'พิษณูโลก':'พิษณุโลก',
                            'ฟินแลนด์':'ไม่ระบุ',
                            'ภาษีเจริญ':'กรุงเทพมหานคร',
                            'ภุเก็ต':'ภูเก็ต',
                            'ภูุเก็ต':'ภูเก็ต',
                            'ภููเก็ต':'ภูเก็ต',
                            'ภูเก็ต':'ภูเก็ต',
                            'ภูเก็็ต':'ภูเก็ต',
                            'ภูเพียง':'ขอนแก่น',
                            'มหาสาคาม':'มหาสารคาม',
                            'มาเลเซีย':'ไม่ระบุ',
                            'มุกาดหาร':'มุกดาหาร',
                            'ยานนาวา':'กรุงเทพมหานคร',
                            'ร.จ.จ.นราธิวาส':'นราธิวาส',
                            'ระบุ':'ไม่ระบุ',
                            'ระยอง\xa0':'ระยอง',
                            'ระอยง':'ระยอง',
                            'ราชบบุรี':'ราชบุรี',
                            'ราชบุรีี':'ราชบุรี',
                            'ราชบุรี\u200b':'ราชบุรี',
                            'ราชุรี':'ราชบุรี',
                            'ราชเทวี':'กรุงเทพมหานคร',
                            'ร้องกวาง':'แพร่',
                            'ลาว':'ไม่ระบุ',
                            'ลบุรี':'ลพบุรี',
                            'ลพบุรี.':'ลพบุรี',
                            'ลพบุรี\xa0':'ลพบุรี',
                            'ลอง':'แพร่',
                            'ลาดกระบัง':'กรุงเทพมหานคร',
                            'ลาดพร้าว':'กรุงเทพมหานคร',
                            'ลาว':'ไม่ระบุ',
                            'ลำพูน\u200b':'ลำพูน',
                            'ลำลูกกา':'กรุงเทพมหานคร',
                            'วังชิ้น':'กรุงเทพมหานคร',
                            'วังทอง':'พิษณุโลก',
                            'วิเศษชัยชาญ':'อ่างทอง',
                            'ศรีษะสะเกษ':'ศรีสะเกษ',
                            'ศรีษะเกษ':'ศรีสะเกษ',
                            'ศรีสระเกษ':'ศรีสะเกษ',
                            'ศรีสะเกษ':'ศรีสะเกษ',
                            'ศีรสะเกษ':'ศรีสะเกษ',
                            'สครสวรรค์':'นครสวรรค์',
                            'สงขลา\u200b':'สงขลา',
                            'สถานที่ทำงาน':'ไม่ระบุ',
                            'สพรรณบุรี':'สุพรรณบุรี',
                            'สมถทรปราการ':'สมุทรปราการ',
                            'สมทรสงคราม':'สมุทรสงคราม',
                            'สมทรสาคร':'สมุทรสาคร',
                            'สมุครปราการ':'สมุทรปราการ',
                            'สมุครสาคร':'สมุทรสาคร',
                            'สมุท':'ไม่ระบุ',
                            'สมุทปราการ':'สมุทรปราการ',
                            'สมุทรปรากร':'สมุทรปราการ',
                            'สมุทรปรากรร':'สมุทรปราการ',
                            'สมุทรปรากา':'สมุทรปราการ',
                            'สมุทรปราการ':'สมุทรปราการ',
                            'สมุทรปราการ\xa0':'สมุทรปราการ',
                            'สมุทรศาคร':'สมุทรสาคร',
                            'สมุทรสงคราม':'สมุทรสงคราม',
                            'สมุทรสาคร':'สมุทรสาคร',
                            'สมุทรสาคร\xa0':'สมุทรสาคร',
                            'สมุทรสาคา':'สมุทรสาคร',
                            'สมุทรสาคาร':'สมุทรสาคร',
                            'สมุทรสาทร':'สมุทรสาคร',
                            'สมุทรสารคร':'สมุทรสาคร',
                            'สมุทราปราการ':'สมุทรปราการ',
                            'สมุทราสาคร':'สมุทรสาคร',
                            'สมุทสงคราม':'สมุทรสงคราม',
                            'สมุทสาคร':'สมุทรสาคร',
                            'สมุย':'สุราษฎร์ธานี',
                            'สมุุทรปราการ':'สมุทรปราการ',
                            'สมุุทรสงคราม':'สมุทรสงคราม',
                            'สมุุทรสาคร':'สมุทรสาคร',
                            'สระก้ว':'สระแก้ว',
                            'สระบุรี':'สระบุรี',
                            'สระบุรี\xa0':'สระบุรี',
                            'สรินทร์':'สุรินทร์',
                            'สวิสเซอร์แลนด์':'ไม่ระบุ',
                            'สอง':'พิษณุโลก',
                            'สองแคว':'พิษณุโลก',
                            'สะพานสูง':'กรุงเทพมหานคร',
                            'สะแก้ว':'สระแก้ว',
                            'สัตหีบ':'ชลบุรี',
                            'สัมผัสผู้ป่วยยืนยัน COVID20':'ไม่ระบุ',
                            'สามุทรสาคร':'สมุทรสาคร',
                            'สารคาม':'มหาสารคาม',
                            'สารภี':'เชียงใหม่',
                            'สิงห์บุรี':'สิงห์บุรี',
                            'สิงห์บุรี\xa0':'สิงห์บุรี',
                            'สุทรปราการ':'สมุทรปราการ',
                            'สุพรรญบุรี':'สุพรรณบุรี',
                            'สุพรรณบุรั':'สุพรรณบุรี',
                            'สุพรรณบุรี':'สุพรรณบุรี',
                            'สุพรรณบุรี\xa0':'สุพรรณบุรี',
                            'สุมทรปราการ':'สมุทรปราการ',
                            'สุมทรสาคร':'สมุทรสาคร',
                            'สุมุทรปราการ':'สมุทรปราการ',
                            'สุมุทรสาคร':'สมุทรสาคร',
                            'สุระบุรี':'สระบุรี',
                            'สุราษ':'สุราษฎร์ธานี',
                            'สุราษฎณ์ธานี':'สุราษฎร์ธานี',
                            'สุราษฎธานี':'สุราษฎร์ธานี',
                            'สุราษฎร์':'สุราษฎร์ธานี',
                            'สุราษฎร์ธานี':'สุราษฎร์ธานี',
                            'สุราษฎร์ฯ':'สุราษฎร์ธานี',
                            'สุราษฎร์์ธานี':'สุราษฎร์ธานี',
                            'สุราษฎ์ธานี':'สุราษฎร์ธานี',
                            'สุราษฏร์ธานี':'สุราษฎร์ธานี',
                            'สุราษธานี':'สุราษฎร์ธานี',
                            'สุราษร์ฎธานี':'สุราษฎร์ธานี',
                            'สุราษร์ธานี':'สุราษฎร์ธานี',
                            'สุราษฯ':'สุราษฎร์ธานี',
                            'สุรินทรื':'สุรินทร์',
                            'สุรินทร์':'สุรินทร์',
                            'สุรินทร์ 32150':'สุรินทร์',
                            'สุรินทร์,รีสะเกษ':'สุรินทร์',
                            'สุรินทร์\xa0':'สุรินทร์',
                            'สุริมทร์':'สุรินทร์',
                            'สุริะแก้ว':'สระแก้ว',
                            'สุวรรณภูมิ':'สมุทรปราการ',
                            'สุุพรรณบุรี':'มหาสารคาม',
                            'สุุราษฎร์ธานี':'สุราษฎร์ธานี',
                            'สุุราษฏร์ธานี':'สุราษฎร์ธานี',
                            'สุุราษฯ':'สุราษฎร์ธานี',
                            'สุุรินทร์':'สุรินทร์',
                            'สุุุโขทัย':'สุโขทัย',
                            'สุโขทัย':'สุโขทัย',
                            'สุโขทัย\xa0':'สุโขทัย',
                            'สูงเม่น':'แพร่',
                            'หนวงบัวลำภู':'หนองบัวลําภู',
                            'หนองงคาย':'หนองคาย',
                            'หนองบัวภู':'หนองบัวลําภู',
                            'หนองบัวลำพูน':'หนองบัวลําภู',
                            'หนองบัวลำภู':'หนองบัวลําภู',
                            'หนองบัวลําภู':'หนองบัวลําภู',
                            'หนองบััวลำภู':'หนองบัวลําภู',
                            'หนองม่วงไข่':'แพร่',
                            'หนอบัวลำภู':'หนองบัวลําภู',
                            'หัวหิน':'ประจวบคีรีขันธ์ ',
                            'ห้วยขวาง':'กรุงเทพมหานคร',
                            'อยุทธยา':'พระนครศรีอยุธยา',
                            'อยุธยา':'พระนครศรีอยุธยา',
                            'อยุุธยา':'พระนครศรีอยุธยา',
                            'อำนาจเจิญ':'อํานาจเจริญ',
                            'อำนาญเจริญ':'อํานาจเจริญ',
                            'อำนาเจริญ':'อํานาจเจริญ',
                            'อิสราเอล':'ไม่ระบุ',
                            'อุดธานี':'อุดรธานี',
                            'อุดร':'อุดรธานี',
                            'อุดรธาณี':'อุดรธานี',
                            'อุดรธานี':'อุดรธานี',
                            'อุดรธานี\xa0':'อุดรธานี',
                            'อุุดรธานีี':'อุดรธานี',
                            'อุดรราชธานี':'อุดรธานี',
                            'อุตรดิตถ์\xa0':'อุตรดิตถ์',
                            'อุตรดิตถ์์':'อุตรดิตถ์',
                            'อุตรดิษถ์':'อุตรดิตถ์',
                            'อุตรดิิตถ์':'อุตรดิตถ์',
                            'อุตรติดถ์':'อุตรดิตถ์',
                            'อุบลราขธานี':'อุบลราชธานี',
                            'อุบลราชธานี\xa0':'อุบลราชธานี',
                            'อุุครดิตถ์':'อุตรดิตถ์',
                            'อุุดรธาน':'อุดรธานี',
                            'อุุตรดิตถ์':'อุตรดิตถ์',
                            'อุุทัยธานี':'อุทัยธานี',
                            'อุุบลราชธานี':'อุบลราชธานี',
                            'อ่งทอง':'อ่างทอง',
                            'อ่างทอง\xa0':'อ่างทอง',
                            'เชิงเทรา':'ฉะเชิงเทรา',
                            'เชีียงราย':'เชียงราย',
                            'เชีียงใหม่':'เชียงใหม่',
                            'เด่นชัย':'แพร่',
                            'เพขรบุรี':'เพชรบูรณ์',
                            'เพขรบูรณื':'เพชรบูรณ์',
                            'เพขรบูรณ์':'เพชรบูรณ์',
                            'เพชรบุรณ์':'เพชรบูรณ์',
                            'เพชรบุรี':'เพชรบูรณ์',
                            'เพชรบุรึ':'เพชรบูรณ์',
                            'เพชรบุุรี':'เพชรบูรณ์',
                            'เพชรบูรณ์\xa0':'เพชรบูรณ์',
                            'เพชรบูร์':'เพชรบูรณ์',
                            'เมือง':'ไม่ระบุ',
                            'เมืองตาก':'ตาก',
                            'เมืองน่าน':'น่าน',
                            'เมืองปัตตานี':'ปัตตานี',
                            'เรือ':'ไม่ระบุ',
                            'เลย\xa0':'เลย',
                            'เวียงจันทร์':'ไม่ระบุ',
                            'เวียงสา':'น่าน',
                            'แม่ทะ':'ลำปาง',
                            'แล้วป่วย':'ไม่ระบุ',
                            'โพธิ์ทอง':'อ่างทอง',
                            'ไชโย':'อ่างทอง',
                            'ไม่มีชื่อหอพัก':'ไม่ระบุ'}, inplace=True)

In [94]:
data_covid['province_of_onset'].replace({'กาญจบุรี':'กาญจนบุรี',
                            'ลำพููน':'ลำพูน',
                            'พัทยา':'ชลบุรี',
                            'อุดรธานีี':'อุดรธานี',
                            'เพชรบุูรณ์':'เพชรบูรณ์',
                            'เพชรบููรณ์':'เพชรบูรณ์',
                            'ประจวบคีรีขันธ์ ':'ประจวบคีรีขันธ์',
                            '\xa0สมุทรสาคร':'สมุทรสาคร',
                            'อุุดรธานี':'อุดรธานี',
                            'กําแพงเพชร':'กำแพงเพชร',
                            'หนองบัวลําภู':'หนองบัวลำภู',
                            'อํานาจเจริญ':'อํานาจเจริญ'}, inplace=True)

In [95]:
set(data_covid['province_of_onset'])

{'กระบี่',
 'กรุงเทพมหานคร',
 'กาญจนบุรี',
 'กาฬสินธุ์',
 'กำแพงเพชร',
 'ขอนแก่น',
 'จันทบุรี',
 'ฉะเชิงเทรา',
 'ชลบุรี',
 'ชัยนาท',
 'ชัยภูมิ',
 'ชุมพร',
 'ตรัง',
 'ตราด',
 'ตาก',
 'นครนายก',
 'นครปฐม',
 'นครพนม',
 'นครราชสีมา',
 'นครศรีธรรมราช',
 'นครสวรรค์',
 'นนทบุรี',
 'นนทุบรี',
 'นราธิวาส',
 'น่าน',
 'บึงกาฬ',
 'บุรีรัมย์',
 'ปทุมธานี',
 'ประจวบคีรีขันธ์',
 'ปราจีนบุรี',
 'ปัตตานี',
 'พระนครศรีอยุธยา',
 'พะเยา',
 'พังงา',
 'พัทลุง',
 'พิจิตร',
 'พิษณุโลก',
 'ภูเก็ต',
 'มหาสารคาม',
 'มุกดาหาร',
 'ยะลา',
 'ยโสธร',
 'ระนอง',
 'ระยอง',
 'ราชบุรี',
 'ร้อยเอ็ด',
 'ลพบุรี',
 'ลำปาง',
 'ลำพูน',
 'ศรีสะเกษ',
 'สกลนคร',
 'สงขลา',
 'สตูล',
 'สมุทรปราการ',
 'สมุทรสงคราม',
 'สมุทรสาคร',
 'สระบุรี',
 'สระแก้ว',
 'สิงห์บุรี',
 'สุพรรณบุรี',
 'สุราษฎร์ธานี',
 'สุรินทร์',
 'สุโขทัย',
 'หนองคาย',
 'หนองบัวลำภู',
 'อำนาจเจริญ',
 'อุดรธานี',
 'อุตรดิตถ์',
 'อุทัยธานี',
 'อุบลราชธานี',
 'อ่างทอง',
 'อํานาจเจริญ',
 'เชียงราย',
 'เชียงใหม่',
 'เพชรบูรณ์',
 'เลย',
 'แพร่',
 'แม่ฮ่องสอน',
 'ไม่ระบุ'}

In [96]:
data_covid

No. announce_date Notified date   sex   age Unit nationality  \
0             1     12/1/2020           NaN  หญิง  61.0   ปี       China   
1             2     17/1/2020           NaN  หญิง  74.0   ปี       China   
2             3     22/1/2020           NaN  หญิง  73.0   ปี    Thailand   
3             4     22/1/2020           NaN   ชาย  68.0   ปี       China   
4             5     24/1/2020           NaN  หญิง  66.0   ปี       China   
...         ...           ...           ...   ...   ...  ...         ...   
291620  3161237     12/3/2022     11/3/2022  หญิง   NaN  NaN         NaN   
291621  3161238     12/3/2022     11/3/2022   ชาย   NaN  NaN   Cambodian   
291622  3161239     12/3/2022     11/3/2022  หญิง   NaN  NaN         NaN   
291623  3161240     12/3/2022     11/3/2022  หญิง   NaN  NaN         NaN   
291624  3161241     12/3/2022     11/3/2022  หญิง   NaN  NaN         NaN   

       province_of_isolation                          risk province_of_onset  \
0              กรุงเทพมหานคร  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
1              กรุงเทพมหานคร  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
2                     นครปฐม  ผู้ที่เดินทางมาจากต่างประเทศ            นครปฐม   
3              กรุงเทพมหานคร  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
4                    นนทบุรี  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
...                      ...                           ...               ...   
291620         กรุงเทพมหานคร              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
291621         กรุงเทพมหานคร              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
291622         กรุงเทพมหานคร              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
291623         กรุงเทพมหานคร              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
291624             เชียงใหม่              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   

       district_of_onset  
0                    NaN  
1                    NaN  
2                  เมือง  
3                    NaN  
4                    NaN  
...                  ...  
291620               NaN  
291621               NaN  
291622               NaN  
291623               NaN  
291624               NaN  

[3184023 rows x 11 columns]

## ตัดเฉพาะ column ที่จะใช้

In [97]:
data_clean = data_covid[['announce_date','sex','risk','province_of_onset']]
data_clean

announce_date   sex                          risk province_of_onset
0          12/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร
1          17/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร
2          22/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ            นครปฐม
3          22/1/2020   ชาย  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร
4          24/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร
...              ...   ...                           ...               ...
291620     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร
291621     12/3/2022   ชาย              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร
291622     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร
291623     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร
291624     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร

[3184023 rows x 4 columns]

In [98]:
data_clean.isnull().any() # เช็คค่า missing 

announce_date        False
sex                  False
risk                 False
province_of_onset    False
dtype: bool

### เลือกภูมิภาคที่ต้องการจะนำไปใช้

ในที่นี่กลุ่มของพวกเราเลือกใช้การแบ่งภูมิภาคแบบการเมือง

In [99]:
data_policy = data_region[['จังหวัด','การเมือง']]
data_policy

จังหวัด            การเมือง
0   อำนาจเจริญ  ตะวันออกเฉียงเหนือ
1       บึงกาฬ  ตะวันออกเฉียงเหนือ
2    บุรีรัมย์  ตะวันออกเฉียงเหนือ
3      ชัยภูมิ  ตะวันออกเฉียงเหนือ
4    กาฬสินธุ์  ตะวันออกเฉียงเหนือ
..         ...                 ...
72       พังงา                 ใต้
73      ภูเก็ต                 ใต้
74       ระนอง                 ใต้
75        สตูล                 ใต้
76        ตรัง                 ใต้

[77 rows x 2 columns]

## รวมตางรางทุกตารางข้อมูล โควิด ภูมิภาค และ เขตสุขภาพ เป็นตารางเดียวกัน

In [100]:
data_covpol = data_clean.merge(data_policy,how='left',left_on='province_of_onset',right_on='จังหวัด') # รวมตาราง data_clean ที่จัดการค่า missing แล้ว นำมาเชื่อมกับ data_policy โดยการต่อในแนวแกน X ด้วยวิธีการ merge
data_covpol

announce_date   sex                          risk province_of_onset  \
0           12/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
1           17/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
2           22/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ            นครปฐม   
3           22/1/2020   ชาย  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
4           24/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
...               ...   ...                           ...               ...   
3184018     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
3184019     12/3/2022   ชาย              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
3184020     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
3184021     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
3184022     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   

               จังหวัด การเมือง  
0        กรุงเทพมหานคร     กลาง  
1        กรุงเทพมหานคร     กลาง  
2               นครปฐม     กลาง  
3        กรุงเทพมหานคร     กลาง  
4        กรุงเทพมหานคร     กลาง  
...                ...      ...  
3184018  กรุงเทพมหานคร     กลาง  
3184019  กรุงเทพมหานคร     กลาง  
3184020  กรุงเทพมหานคร     กลาง  
3184021  กรุงเทพมหานคร     กลาง  
3184022  กรุงเทพมหานคร     กลาง  

[3184023 rows x 6 columns]

In [101]:
data = data_covpol.merge(data_mentalhealth,how='left',left_on='จังหวัด',right_on='จังหวัด') # รวมตาราง data_covpol ที่จัดการค่า missing แล้ว นำมาเชื่อมกับ data_mentalhealth โดยการต่อในแนวแกน X ด้วยวิธีการ merge
data

announce_date   sex                          risk province_of_onset  \
0           12/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
1           17/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
2           22/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ            นครปฐม   
3           22/1/2020   ชาย  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
4           24/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
...               ...   ...                           ...               ...   
3279747     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
3279748     12/3/2022   ชาย              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
3279749     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
3279750     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
3279751     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   

               จังหวัด การเมือง          เขตสุขภาพ  
0        กรุงเทพมหานคร     กลาง  เขตสุขภาพส่วนกลาง  
1        กรุงเทพมหานคร     กลาง  เขตสุขภาพส่วนกลาง  
2               นครปฐม     กลาง     เขตสุขภาพที่ 5  
3        กรุงเทพมหานคร     กลาง  เขตสุขภาพส่วนกลาง  
4        กรุงเทพมหานคร     กลาง  เขตสุขภาพส่วนกลาง  
...                ...      ...                ...  
3279747  กรุงเทพมหานคร     กลาง  เขตสุขภาพส่วนกลาง  
3279748  กรุงเทพมหานคร     กลาง  เขตสุขภาพส่วนกลาง  
3279749  กรุงเทพมหานคร     กลาง  เขตสุขภาพส่วนกลาง  
3279750  กรุงเทพมหานคร     กลาง  เขตสุขภาพส่วนกลาง  
3279751  กรุงเทพมหานคร     กลาง  เขตสุขภาพส่วนกลาง  

[3279752 rows x 7 columns]

**เปลี่ยนชื่อคอลัมน์จาก การเมือง เป็น ภูมิภาค**

In [102]:
data.rename(columns={'การเมือง':'ภูมิภาค'},inplace=True)

In [103]:
data

announce_date   sex                          risk province_of_onset  \
0           12/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
1           17/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
2           22/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ            นครปฐม   
3           22/1/2020   ชาย  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
4           24/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
...               ...   ...                           ...               ...   
3279747     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
3279748     12/3/2022   ชาย              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
3279749     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
3279750     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
3279751     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   

               จังหวัด ภูมิภาค          เขตสุขภาพ  
0        กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
1        กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
2               นครปฐม    กลาง     เขตสุขภาพที่ 5  
3        กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
4        กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
...                ...     ...                ...  
3279747  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
3279748  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
3279749  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
3279750  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
3279751  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  

[3279752 rows x 7 columns]

In [104]:
data.isnull().sum() # ดูว่ามีค่า missing กี่ตัว

announce_date             0
sex                       0
risk                      0
province_of_onset         0
จังหวัด              118952
ภูมิภาค              118952
เขตสุขภาพ            146675
dtype: int64

เนื่องจากค่า missing ที่เกิดขึ้นมาจากที่ข้อมูลจาก data['province_of_onset'] ว่า ไม่ระะบุ ทำให้ไม่สามารถนำมาเชื่อมกับตารางเขตสุขภาพจาก data_mentalhealth['จังหวัด'] ได้ จึงเกิดค่า NaN ขึ้น ดังนั้นจะทำการ drop ค่าทิ้งไป

In [105]:
data_project = data.dropna() # ลบค่า missing
data_project

announce_date   sex                          risk province_of_onset  \
0           12/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
1           17/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
2           22/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ            นครปฐม   
3           22/1/2020   ชาย  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
4           24/1/2020  หญิง  ผู้ที่เดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
...               ...   ...                           ...               ...   
3279747     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
3279748     12/3/2022   ชาย              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
3279749     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
3279750     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   
3279751     12/3/2022  หญิง              ทัณฑสถาน/เรือนจำ     กรุงเทพมหานคร   

               จังหวัด ภูมิภาค          เขตสุขภาพ  
0        กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
1        กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
2               นครปฐม    กลาง     เขตสุขภาพที่ 5  
3        กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
4        กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
...                ...     ...                ...  
3279747  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
3279748  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
3279749  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
3279750  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
3279751  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  

[3133077 rows x 7 columns]

In [106]:
data_project.isnull().any() # เช็คดูค่า missing

announce_date        False
sex                  False
risk                 False
province_of_onset    False
จังหวัด              False
ภูมิภาค              False
เขตสุขภาพ            False
dtype: bool

***เนื่องจากไฟล์ data_project นั้นมีข้อมูลมากเกินไป 3133077 rows ทำให้ไม่สามารถนำข้อมูลไปใช้ทำ Dashboard ได้ ดังนั้นตัดข้อมูลออกเป็น 8 ไฟล์ ไฟล์ที่ 1 - 7 ไฟล์ละ 400000 rows และไฟล์ที่ 8 เป็น 333077 rows***

In [107]:
data_dashboard1 = data_project.iloc[:400000,:]
data_dashboard1

announce_date   sex                                      risk  \
0          12/1/2020  หญิง              ผู้ที่เดินทางมาจากต่างประเทศ   
1          17/1/2020  หญิง              ผู้ที่เดินทางมาจากต่างประเทศ   
2          22/1/2020  หญิง              ผู้ที่เดินทางมาจากต่างประเทศ   
3          22/1/2020   ชาย              ผู้ที่เดินทางมาจากต่างประเทศ   
4          24/1/2020  หญิง              ผู้ที่เดินทางมาจากต่างประเทศ   
...              ...   ...                                       ...   
433447     21/7/2021  หญิง  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
433448     21/7/2021  หญิง  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
433449     21/7/2021   ชาย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
433450     21/7/2021   ชาย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
433451     21/7/2021   ชาย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   

       province_of_onset        จังหวัด ภูมิภาค          เขตสุขภาพ  
0          กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
1          กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
2                 นครปฐม         นครปฐม    กลาง     เขตสุขภาพที่ 5  
3          กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
4          กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
...                  ...            ...     ...                ...  
433447           นนทบุรี        นนทบุรี    กลาง     เขตสุขภาพที่ 4  
433448           นนทบุรี        นนทบุรี    กลาง     เขตสุขภาพที่ 4  
433449           นนทบุรี        นนทบุรี    กลาง     เขตสุขภาพที่ 4  
433450           นนทบุรี        นนทบุรี    กลาง     เขตสุขภาพที่ 4  
433451           นนทบุรี        นนทบุรี    กลาง     เขตสุขภาพที่ 4  

[400000 rows x 7 columns]

In [108]:
data_dashboard2 = data_project.iloc[400000:800000,:]
data_dashboard2

announce_date   sex                                      risk  \
433452     21/7/2021   ชาย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
433453     21/7/2021   ชาย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
433454     21/7/2021   ชาย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
433455     21/7/2021   ชาย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
433456     21/7/2021   ชาย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
...              ...   ...                                       ...   
864899     12/8/2021   ชาย                            ไปสถานที่ชุมชน   
864900     12/8/2021   ชาย                            ไปสถานที่ชุมชน   
864901     12/8/2021  หญิง                            ไปสถานที่ชุมชน   
864902     12/8/2021  หญิง  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
864903     12/8/2021  หญิง  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   

       province_of_onset  จังหวัด ภูมิภาค       เขตสุขภาพ  
433452           นนทบุรี  นนทบุรี    กลาง  เขตสุขภาพที่ 4  
433453           นนทบุรี  นนทบุรี    กลาง  เขตสุขภาพที่ 4  
433454           นนทบุรี  นนทบุรี    กลาง  เขตสุขภาพที่ 4  
433455           นนทบุรี  นนทบุรี    กลาง  เขตสุขภาพที่ 4  
433456           นนทบุรี  นนทบุรี    กลาง  เขตสุขภาพที่ 4  
...                  ...      ...     ...             ...  
864899           ราชบุรี  ราชบุรี    กลาง  เขตสุขภาพที่ 5  
864900           ราชบุรี  ราชบุรี    กลาง  เขตสุขภาพที่ 5  
864901           ราชบุรี  ราชบุรี    กลาง  เขตสุขภาพที่ 5  
864902           ราชบุรี  ราชบุรี    กลาง  เขตสุขภาพที่ 5  
864903           ราชบุรี  ราชบุรี    กลาง  เขตสุขภาพที่ 5  

[400000 rows x 7 columns]

In [109]:
data_dashboard3 = data_project.iloc[800000:1200000,:]
data_dashboard3

announce_date   sex                                      risk  \
864904      12/8/2021  หญิง  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
864905      12/8/2021  หญิง                                    อื่น ๆ   
864906      12/8/2021   ชาย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
864907      12/8/2021  หญิง                                    อื่น ๆ   
864908      12/8/2021   ชาย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
...               ...   ...                                       ...   
1294904      3/9/2021  หญิง                      อยู่ระหว่างการสอบสวน   
1294905      3/9/2021  หญิง                      อยู่ระหว่างการสอบสวน   
1294906      3/9/2021  หญิง                      อยู่ระหว่างการสอบสวน   
1294907      3/9/2021  หญิง                      อยู่ระหว่างการสอบสวน   
1294908      3/9/2021  หญิง                      อยู่ระหว่างการสอบสวน   

        province_of_onset        จังหวัด ภูมิภาค          เขตสุขภาพ  
864904            ราชบุรี        ราชบุรี    กลาง     เขตสุขภาพที่ 5  
864905            ราชบุรี        ราชบุรี    กลาง     เขตสุขภาพที่ 5  
864906            ราชบุรี        ราชบุรี    กลาง     เขตสุขภาพที่ 5  
864907            ราชบุรี        ราชบุรี    กลาง     เขตสุขภาพที่ 5  
864908            ราชบุรี        ราชบุรี    กลาง     เขตสุขภาพที่ 5  
...                   ...            ...     ...                ...  
1294904     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
1294905     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
1294906     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
1294907     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
1294908     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  

[400000 rows x 7 columns]

In [110]:
data_dashboard4 = data_project.iloc[1200000:1600000,:]
data_dashboard4

announce_date   sex                                              risk  \
1294909      3/9/2021  หญิง                              อยู่ระหว่างการสอบสวน   
1294910      3/9/2021  หญิง                              อยู่ระหว่างการสอบสวน   
1294911      3/9/2021  หญิง                              อยู่ระหว่างการสอบสวน   
1294912      3/9/2021  หญิง                              อยู่ระหว่างการสอบสวน   
1294913      3/9/2021  หญิง                              อยู่ระหว่างการสอบสวน   
...               ...   ...                                               ...   
1710608     4/10/2021   ชาย  การค้นหาผู้ป่วยเชิงรุกและค้นหาผู้ติดเชื้อในชุมชน   
1710609     4/10/2021   ชาย  การค้นหาผู้ป่วยเชิงรุกและค้นหาผู้ติดเชื้อในชุมชน   
1710610     4/10/2021  หญิง  การค้นหาผู้ป่วยเชิงรุกและค้นหาผู้ติดเชื้อในชุมชน   
1710611     4/10/2021  หญิง  การค้นหาผู้ป่วยเชิงรุกและค้นหาผู้ติดเชื้อในชุมชน   
1710612     4/10/2021   ชาย  การค้นหาผู้ป่วยเชิงรุกและค้นหาผู้ติดเชื้อในชุมชน   

        province_of_onset        จังหวัด ภูมิภาค          เขตสุขภาพ  
1294909     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
1294910     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
1294911     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
1294912     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
1294913     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
...                   ...            ...     ...                ...  
1710608        ปราจีนบุรี     ปราจีนบุรี    กลาง     เขตสุขภาพที่ 6  
1710609        ปราจีนบุรี     ปราจีนบุรี    กลาง     เขตสุขภาพที่ 6  
1710610        ปราจีนบุรี     ปราจีนบุรี    กลาง     เขตสุขภาพที่ 6  
1710611        ปราจีนบุรี     ปราจีนบุรี    กลาง     เขตสุขภาพที่ 6  
1710612        ปราจีนบุรี     ปราจีนบุรี    กลาง     เขตสุขภาพที่ 6  

[400000 rows x 7 columns]

In [111]:
data_dashboard5 = data_project.iloc[1600000:2000000,:]
data_dashboard5

announce_date   sex                                              risk  \
1710613     4/10/2021  หญิง  การค้นหาผู้ป่วยเชิงรุกและค้นหาผู้ติดเชื้อในชุมชน   
1710614     4/10/2021   ชาย                              อยู่ระหว่างการสอบสวน   
1710615     4/10/2021  หญิง                              อยู่ระหว่างการสอบสวน   
1710616     4/10/2021  หญิง          สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
1710617     4/10/2021   ชาย          สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
...               ...   ...                                               ...   
2115676    16/11/2021  หญิง                                            อื่น ๆ   
2115677    16/11/2021  หญิง          สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
2115678    16/11/2021  หญิง                                            อื่น ๆ   
2115679    16/11/2021  หญิง                                            อื่น ๆ   
2115680    16/11/2021  หญิง          สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   

        province_of_onset        จังหวัด ภูมิภาค          เขตสุขภาพ  
1710613        ปราจีนบุรี     ปราจีนบุรี    กลาง     เขตสุขภาพที่ 6  
1710614        ปราจีนบุรี     ปราจีนบุรี    กลาง     เขตสุขภาพที่ 6  
1710615        ปราจีนบุรี     ปราจีนบุรี    กลาง     เขตสุขภาพที่ 6  
1710616        ปราจีนบุรี     ปราจีนบุรี    กลาง     เขตสุขภาพที่ 6  
1710617        ปราจีนบุรี     ปราจีนบุรี    กลาง     เขตสุขภาพที่ 6  
...                   ...            ...     ...                ...  
2115676     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
2115677     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
2115678     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
2115679     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
2115680     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  

[400000 rows x 7 columns]

In [112]:
data_dashboard6 = data_project.iloc[2000000:2400000,:]
data_dashboard6

announce_date   sex                                      risk  \
2115681    16/11/2021  หญิง                            ไปสถานที่ชุมชน   
2115682    16/11/2021  หญิง  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
2115683    16/11/2021   ชาย                                    อื่น ๆ   
2115684    16/11/2021  หญิง  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
2115685    16/11/2021   ชาย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
...               ...   ...                                       ...   
2521088     30/1/2022   ชาย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
2521089     30/1/2022  หญิง  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
2521090     30/1/2022   ชาย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
2521091     30/1/2022   ชาย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
2521092     30/1/2022   ชาย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   

        province_of_onset        จังหวัด ภูมิภาค          เขตสุขภาพ  
2115681     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
2115682     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
2115683     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
2115684          ปทุมธานี       ปทุมธานี    กลาง     เขตสุขภาพที่ 4  
2115685     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
...                   ...            ...     ...                ...  
2521088            พิจิตร         พิจิตร   เหนือ     เขตสุขภาพที่ 3  
2521089            พิจิตร         พิจิตร   เหนือ     เขตสุขภาพที่ 3  
2521090         อุทัยธานี      อุทัยธานี   เหนือ     เขตสุขภาพที่ 3  
2521091         อุทัยธานี      อุทัยธานี   เหนือ     เขตสุขภาพที่ 3  
2521092         อุทัยธานี      อุทัยธานี   เหนือ     เขตสุขภาพที่ 3  

[400000 rows x 7 columns]

In [113]:
data_dashboard7 = data_project.iloc[2400000:2800000,:]
data_dashboard7

announce_date   sex                                      risk  \
2521093     30/1/2022   ชาย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
2521094     30/1/2022  หญิง  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
2521095     30/1/2022   ชาย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
2521096     30/1/2022  หญิง  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
2521097     30/1/2022  หญิง  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
...               ...   ...                                       ...   
2933889     26/2/2022   ชาย                                    อื่น ๆ   
2933890     26/2/2022   ชาย                                    อื่น ๆ   
2933891     26/2/2022  หญิง                                    อื่น ๆ   
2933892     26/2/2022   ชาย                                    อื่น ๆ   
2933893     26/2/2022   ชาย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   

        province_of_onset        จังหวัด ภูมิภาค          เขตสุขภาพ  
2521093         อุทัยธานี      อุทัยธานี   เหนือ     เขตสุขภาพที่ 3  
2521094         อุทัยธานี      อุทัยธานี   เหนือ     เขตสุขภาพที่ 3  
2521095         อุทัยธานี      อุทัยธานี   เหนือ     เขตสุขภาพที่ 3  
2521096         อุทัยธานี      อุทัยธานี   เหนือ     เขตสุขภาพที่ 3  
2521097         อุทัยธานี      อุทัยธานี   เหนือ     เขตสุขภาพที่ 3  
...                   ...            ...     ...                ...  
2933889         เพชรบูรณ์      เพชรบูรณ์   เหนือ     เขตสุขภาพที่ 2  
2933890     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
2933891     กรุงเทพมหานคร  กรุงเทพมหานคร    กลาง  เขตสุขภาพส่วนกลาง  
2933892            ชัยนาท         ชัยนาท    กลาง     เขตสุขภาพที่ 3  
2933893            นครปฐม         นครปฐม    กลาง     เขตสุขภาพที่ 5  

[400000 rows x 7 columns]

In [114]:
data_dashboard8 = data_project.iloc[2800000:,:]
data_dashboard8

announce_date   sex                                      risk  \
2933894     26/2/2022  หญิง                                    อื่น ๆ   
2933895     26/2/2022   ชาย                                    อื่น ๆ   
2933897     26/2/2022  หญิง  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
2933902     26/2/2022   ชาย                                    อื่น ๆ   
2933927     26/2/2022   ชาย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้า   
...               ...   ...                                       ...   
3279747     12/3/2022  หญิง                          ทัณฑสถาน/เรือนจำ   
3279748     12/3/2022   ชาย                          ทัณฑสถาน/เรือนจำ   
3279749     12/3/2022  หญิง                          ทัณฑสถาน/เรือนจำ   
3279750     12/3/2022  หญิง                          ทัณฑสถาน/เรือนจำ   
3279751     12/3/2022  หญิง                          ทัณฑสถาน/เรือนจำ   

        province_of_onset        จังหวัด             ภูมิภาค  \
2933894               ตาก            ตาก               เหนือ   
2933895     กรุงเทพมหานคร  กรุงเทพมหานคร                กลาง   
2933897     กรุงเทพมหานคร  กรุงเทพมหานคร                กลาง   
2933902     กรุงเทพมหานคร  กรุงเทพมหานคร                กลาง   
2933927            สกลนคร         สกลนคร  ตะวันออกเฉียงเหนือ   
...                   ...            ...                 ...   
3279747     กรุงเทพมหานคร  กรุงเทพมหานคร                กลาง   
3279748     กรุงเทพมหานคร  กรุงเทพมหานคร                กลาง   
3279749     กรุงเทพมหานคร  กรุงเทพมหานคร                กลาง   
3279750     กรุงเทพมหานคร  กรุงเทพมหานคร                กลาง   
3279751     กรุงเทพมหานคร  กรุงเทพมหานคร                กลาง   

                 เขตสุขภาพ  
2933894     เขตสุขภาพที่ 1  
2933895  เขตสุขภาพส่วนกลาง  
2933897  เขตสุขภาพส่วนกลาง  
2933902  เขตสุขภาพส่วนกลาง  
2933927     เขตสุขภาพที่ 8  
...                    ...  
3279747  เขตสุขภาพส่วนกลาง  
3279748  เขตสุขภาพส่วนกลาง  
3279749  เขตสุขภาพส่วนกลาง  
3279750  เขตสุขภาพส่วนกลาง  
3279751  เขตสุขภาพส่วนกลาง  

[333077 rows x 7 columns]

# Save file ออกมาเพื่อนำไปใช้ทำ Dashboad

In [115]:
data_dashboard1.to_csv('data_dashboard1.csv', encoding='utf-8')

In [116]:
data_dashboard2.to_csv('data_dashboard2.csv', encoding='utf-8')

In [117]:
data_dashboard3.to_csv('data_dashboard3.csv', encoding='utf-8')

In [118]:
data_dashboard4.to_csv('data_dashboard4.csv', encoding='utf-8')

In [119]:
data_dashboard5.to_csv('data_dashboard5.csv', encoding='utf-8')

In [120]:
data_dashboard6.to_csv('data_dashboard6.csv', encoding='utf-8')

In [121]:
data_dashboard7.to_csv('data_dashboard7.csv', encoding='utf-8')

In [122]:
data_dashboard8.to_csv('data_dashboard8.csv', encoding='utf-8')